src/graphql/node/control.rs

```diff
@@ -145,20 +168,25 @@ pub struct ApplyNodeResponse {
 
 async fn notify_agents(
     agent_manager: &BoxedAgentManager,
-    update_agent_keys: &[&str],
-    disable_agent_keys: &[&str],
+    hostname: &str,
+    update_agent_ids: &[&str],
+    disable_agent_ids: &[&str],
 ) -> Result<()> {
-    let update_futures = update_agent_keys.iter().map(|agent_key| async move {
-        agent_manager.update_config(agent_key).await.map_err(|e| {
-            async_graphql::Error::new(format!(
-                "Failed to notify agent for config update {agent_key}: {e}"
-            ))
-        })
+    let update_futures = update_agent_ids.iter().map(|agent_id| async move {
+        let agent_key = format!("{agent_id}@{hostname}");
+        agent_manager
+            .update_config(agent_key.as_str())
+            .await
+            .map_err(|e| {
+                async_graphql::Error::new(format!(
+                    "Failed to notify agent for config update {agent_key}: {e}"
+                ))
+            })
     });
 
     // TODO: We need to implement the logic to disable agents. For now, we will only log the
     // message.
-    info!("Agents {disable_agent_keys:?} need to be notified to be disabled, but disabling logic is not yet implemented");
+    info!("Agents {disable_agent_ids:?} need to be notified to be disabled, but disabling logic is not yet implemented");
```

Comment:
The second sentence doesn’t seem necessary to me. Since log messages are visible to outside partners and even users, I think it's inappropriate to reveal anything that should have been implemented.

Reply:
Done